### DISCUSSION TITLE
Support for Memory Optimized SQL Server database and tables

### DISCUSSION THREAD

**Discussion by marco-gemdat** [fluentmigrator/fluentmigrator#1611] [Q&A]
Hi

Does the framework currently support memory optimized SQL server databases and tables?
See https://docs.microsoft.com/en-us/sql/relational-databases/in-memory-oltp/overview-and-usage-scenarios?view=sql-server-ver16

**Comment by jzabroski** [comment]
Currently, table expressions don't have "`ISupportAdditionalFeatures`", so no.  `ISupportAdditionalFeatures` is how each database provider/generator can deviate from the SQL interface, all databases FluentMigrator supports, have in common.

I don't know why table expressions don't have this interface.  Arguably, they should. @fubar-coder  if you have anything to add, please add in the comments.

# Relevant Interfaces

1. https://github.com/fluentmigrator/fluentmigrator/blob/9c87d698b161b3a1bb596a5624c02891a64459c5/src/FluentMigrator.Abstractions/Infrastructure/ISupportAdditionalFeatures.cs
2. https://github.com/fluentmigrator/fluentmigrator/blob/9c87d698b161b3a1bb596a5624c02891a64459c5/src/FluentMigrator.Abstractions/Infrastructure/Extensions/AdditionalFeaturesExtensions.cs
3. https://github.com/fluentmigrator/fluentmigrator/blob/9c87d698b161b3a1bb596a5624c02891a64459c5/src/FluentMigrator.Abstractions/Expressions/CreateTableExpression.cs#L31 - note that it doesn't implement ISupportAdditionalFeatures

# Implementing additional feature validation
1. https://github.com/fluentmigrator/fluentmigrator/blob/9c87d698b161b3a1bb596a5624c02891a64459c5/src/FluentMigrator.Abstractions/ValidationUtilities.cs

# Implementing Generator Logic for `ISupportAdditionalFeatures`
1. https://github.com/fluentmigrator/fluentmigrator/blob/9c87d698b161b3a1bb596a5624c02891a64459c5/src/FluentMigrator.Runner.SqlServer/Generators/SqlServer/SqlServer2005Generator.cs
2. https://github.com/fluentmigrator/fluentmigrator/blob/9c87d698b161b3a1bb596a5624c02891a64459c5/src/FluentMigrator.Runner.SqlServer/Generators/SqlServer/SqlServer2008Generator.cs

# Examples
1. Online index creation: https://github.com/fluentmigrator/fluentmigrator/blob/9c87d698b161b3a1bb596a5624c02891a64459c5/src/FluentMigrator.Extensions.SqlServer/SqlServer/SqlServerExtensions.Online.cs
5. Included Columns in Index: https://github.com/fluentmigrator/fluentmigrator/blob/9c87d698b161b3a1bb596a5624c02891a64459c5/src/FluentMigrator.Extensions.SqlServer/SqlServer/SqlServerExtensions.Include.cs
6. Sparse Columns: https://github.com/fluentmigrator/fluentmigrator/blob/9c87d698b161b3a1bb596a5624c02891a64459c5/src/FluentMigrator.Extensions.SqlServer/SqlServer/SqlServerExtensions.cs#L79-L85
7. Data Compression: https://github.com/fluentmigrator/fluentmigrator/blob/9c87d698b161b3a1bb596a5624c02891a64459c5/src/FluentMigrator.Extensions.SqlServer/SqlServer/SqlServerExtensions.DataCompression.cs

**Reply by fubar-coder** [reply]
@jzabroski There's nothing against having this interface in this place. I probably never needed it and therefore didn't implement it.

**Reply by jzabroski** [reply]
@marco-gemdat  I'll add this interface to create table syntax for v4.0 release.
